javascript/tables/src/readTable.utils.ts

```diff
@@ -39,21 +39,20 @@ export function getParser(options?: ParserOptions) {
 function arqueroParser(text: string, options?: ParserOptions): ColumnTable {
 	const mappedOptions = mapProps(ParserType.Arquero, options) as CSVParseOptions
 	let table = fromCSV(text, mappedOptions)
-	if (options?.nRows) {
-		const subset = nRows([...table], options.nRows)
-		table = from(subset)
-	}
-	return table
+	return options?.readRows ? from([...table].slice(0, options.readRows)) : table
```

Comment:
Is it necessary to copy this table array here with `[...table]`? Since it is a private variable you don't need to work about mutation.

Reply:
Oh, I think you are still adding more than needed here anyway - my previous comment should work where you create the table once and then return either a slice on it or the whole thing